compiler-rt/test/ubsan/TestCases/TypeCheck/misaligned.cpp

```diff
@@ -47,6 +50,16 @@ int main(int, char **argv) {
     return *p && 0;
     // CHECK-STACK-LOAD: #0 {{.*}}main{{.*}}misaligned.cpp
 
+  case 'L': {
+    int x;
+    // CHECK-MEMCPY-LOAD: misaligned.cpp:[[#@LINE+4]]{{(:16)?}}: runtime error: load of misaligned address [[PTR:0x[0-9a-f]*]] for type 'const void *', which requires 4 byte alignment
```

Comment:
'const void *' is wrong here :(

Reply:
Maybe not wrong, but unhelpful.
Issue is that 'int*' was misaligned, not that we pass it into void*.

Reply:
Yes, the issue was mentioned in the description
```
The error message refers to void * instead of int *, which is not perfect
but should be acceptable.

runtime error: store to misaligned address [[PTR:0x[0-9a-f]*]] for type 'void *'
```

Anyhow, `BI__builtin_HEXAGON_V6_vmaskedstoreq` strips an implicit cast. I added a similar mechanism in 9b705127b17513ee17e76fb903e85500c832d8aa